lib/query.js

```diff
@@ -2248,6 +2247,9 @@ Query.prototype.error = function error(err) {
  */
 
 Query.prototype._unsetCastError = function _unsetCastError() {
+  if (this._error == null) {
+    return;
+  }
   if (this._error != null && !(this._error instanceof CastError)) {
```

Comment:
@vkarpov15  Since you already checked `(this._error == null` above, you can simplify the condition as below. 
`if (!(this._error instanceof CastError)) {}`

Reply:
See #15104 :+1: